pkgs/development/guile-modules/bytestructures/default.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchFromGitHub, guile
+, autoreconfHook, pkg-config
+}:
+
+stdenv.mkDerivation rec {
+  pname = "scheme-bytestructures";
+  version = "1.0.7";
+
+  src = fetchFromGitHub {
+    owner = "TaylanUB";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0q0habjiy3h9cigb7q1br9kz6z212dn2ab31f6dgd3rrmsfn5rvb";
+  };
+
+  postConfigure = ''
+    sed -i '/moddir\s*=/s%=.*%=''${out}/share/guile/site%' Makefile;
```

Comment:
Did you try something like this:

https://github.com/NixOS/nixpkgs/blob/fa0849d02308ea4d0b646f6fffff07cd042ab810/pkgs/development/guile-modules/guile-ncurses/default.nix#L17-L19

Or perhaps even better:

```nix
configureFlags = [
  "--with-guilesitedir=${placeholder "out"}/share/guile/site"
];

Reply:
If that doesn't work I'd advice contacting upstream.

Reply:
Yes, like I said to timo above, it's not something that can be substituted by autotools, it's hardcoded in the Makefile.

Reply:
And what's wrong with this output:

```
$ tree $(nix-build -A guile-json)
result
├── lib
│   └── guile
│       └── 2.2
│           └── site-ccache
│               ├── json
│               │   ├── builder.go
│               │   └── parser.go
│               └── json.go
└── share
    └── guile
        └── site
            └── 2.2
                ├── json
                │   ├── builder.scm
                │   └── parser.scm
                └── json.scm

10 directories, 6 files
```

Reply:
This is what you get if you don't specify any configure flags nor apply that postConfigure hook.

Reply:
These paths have to be linked at by guile, for every package needed by guix. Note that some of these packages by default do not have 2.2 in that path, some don't even properly specify their paths (and therefore end up trying to write directly to the guile derivation), or have extensions and try and write those somewhere that they won't be found. For the sake of consistency across guile modules and sanity of everyone trying to build this, I thought it might be nice to fix those

Reply:
> These paths have to be linked at by guile

Are you talking about the wrapping done at `guix/default.nix` that adds `GUILE_LOAD_COMPILED_PATH` to the environment?

It just seems weird to me, that a package maintained by Guix's developers (at least in a certain sense) - a package that's needed to bootstrap a package manager which is based on Nix, will need such hacky workarounds to fix paths in `$out`. Guix doesn't put these to any of those guile dependencies, see:

- https://git.savannah.gnu.org/cgit/guix.git/tree/gnu/packages/guile.scm?id=5d9e2187929ed7e8d46ec3cb3174fd78c1846360#n490
- https://git.savannah.gnu.org/cgit/guix.git/tree/gnu/packages/guile.scm?id=5d9e2187929ed7e8d46ec3cb3174fd78c1846360#n654

Moreover, they have Guile 3.0.